### ISSUE TITLE
Got 200 return code when updating device profile by PUT method with non-existed deviceprofile id

### ISSUE THREAD

**Issue by cherrycl** [edgexfoundry/edgex-go#1632]
# 🐞 Bug Report

### Affected Services

 The issue is located in: Metadata

### Is this a regression? Yes

 Yes, the previous version in which this bug was not present was: ....

### Description and Minimal Reproduction
1. Startup EdgeX
2. Put /api/v1/deviceprofile with the following body.
> Method: PUT
> URL: http://localhost:48081/api/v1/deviceprofile
> {
>     "id": "57f4995be4b060e268aeb000",
>     "description": "Test DeviceProfile after PUT",
>     "coreCommands": [
>       {
>         "id": "57f4995be4b060e268aeb084",
>         "name": "cooling",
>         "get": {
>           "response": {
>             "errorDescription": "Integer Value",
>             "expectedValues": "temperature"
>           }
>         },
>         "put": {
>           "response": {
>             "errorDescription": "Integer Value",
>             "expectedValues": "temperature"
>           }
>         }
>       }
>     ]
> }

Expected result: return code 404

## 🔥 Exception or Error
<pre><code>
HTTP/1.1 200
status: 200
Content-Type: application/json
Date: Wed, 07 Aug 2019 02:42:26 GMT
Content-Length: 4
true

Metadata log shows as below.
level=ERROR ts=2019-08-07T02:34:20.977017891Z app=edgex-core-metadata source=rest_deviceprofile.go:444 msg="Item not found"
level=ERROR ts=2019-08-07T02:34:21.08602157Z app=edgex-core-metadata source=rest_deviceprofile.go:145 msg="device profile not found -- id: '{{deviceprofileId}}' name: ''"

</code></pre>

## 🌍  Your Environment
**Deployment Environment:** ubuntu 18.04 Docker container

**EdgeX Version:** 1.1.0

**Anything else relevant?**

**Comment by difince** [comment]
Hi @cherrycl
I was not able to reproduce the problem.
Update of a device profile with not-existing id and name -> returns 404 - Not Found as it is expected.

Note: there is some mismatch between the request you have been posted (it contains "id": "57f4995be4b060e268aeb000"),and the metadata log you provide - it contains `{{deviceprofileId}} `for device profile id.
`level=ERROR ts=2019-08-07T02:34:21.08602157Z app=edgex-core-metadata source=rest_deviceprofile.go:145 msg="device profile not found -- id: '{{deviceprofileId}}' name: ''"`
I guess the log is from another request, probably coming form postman? 

Could you verify the issue with just executing pure curl request, so you will be sure that you see the correct output 
`curl -X PUT  http://localhost:48081/api/v1/deviceprofile -d @/tmp/file_containg_the_Dev_Profile -v `

Note: I have tested with commit 1175e1ff989bcb6172aca90048298ba7681bad4c

**Comment by cherrycl** [comment]
Hi @difince,
Thanks for the investigating.
I found the issue on Jenkins BlackBox job and did verification from the postman. The log generated when sending put API from postman. 

I found the Blackbox test job was still failed on the last run, so manually verified the issue on the latest image (image digests is as below), the problem did not exist.
`"RepoDigests": [ "nexus3.edgexfoundry.org:10004/docker-core-metadata-go@sha256:f2f84b944ecff73610bb056d91995275c7befe43b95e46a3545e5d009333ee3d"
`
I will close this issue after blackbox testcase(deviceprofile_error_4xx / 162) pass by jenkins job.
